lightningd/peer_control.c

```diff
@@ -638,7 +638,20 @@ static struct peer_connected_hook_response *
 peer_connected_deserialize(const tal_t *ctx, const char *buffer,
 			   const jsmntok_t *toks)
 {
-	return NULL;
+	const jsmntok_t *resulttok;
+	struct peer_connected_hook_response *resp;
+	resulttok = json_get_member(buffer, toks, "result");
+	if (!resulttok) {
+		return NULL;
+	}
+
+	resp = tal(ctx, struct peer_connected_hook_response);
+	if (json_tok_streq(buffer, resulttok, "continue"))
+		resp->result = peer_connected_continue;
+	else
+		resp->result = peer_connected_disconnect;
```

Comment:
Probably best to be explicit about the alternative to continue, eg "disconnect"?  Otherwise typos can have bad effects, and its hard to add other return types in the future (eg. an error message to send or something?).